tensorflow/core/lib/gtl/flatrep.h

```diff
@@ -339,7 +339,7 @@ class FlatRep {
 
   inline size_t NextIndex(size_t i, uint32 num_probes) const {
     // Quadratic probing.
-    return (i + num_probes) & mask_;
```

Comment:
the <i> that is passed here is the current index in the probe process, not the original index.  So I think it is quadratic (though without an extra factor), since the distance from original at step <s> is 1+2+3+...+(S-1)+S, which is S*(S+1)/2.

Reply:
Got it, a good algorithm.